src/Multisig.sol

```diff
@@ -96,10 +96,26 @@ contract Multisig {
     /// @param data Transaction data
     /// @return pendingHash for the created tx
     function createTx(address to, uint value, bytes memory data, uint nNeeded) public onlyOwner returns (bytes32) {
+
         // create Transaction
+        bytes32 txHash = keccak256(abi.encodePacked(to, value, data));
+        txs[txHash] = Transaction({ to: to, value: value, data: data });
+
         // create PendingTx
-        // signTx
+        bytes32 pendingHash = keccak256(abi.encodePacked(txHash, nNeeded, block.number));
+        PendingTx storage pendingTx = pending[pendingHash];
+        pendingTx.txHash = txHash;
+        pendingTx.nNeeded = nNeeded;
+        pendingTx.nSigned = 0;
+
         // log event
+        emit CreateTx(to, value, data, txHash, pendingHash);
+
+        // signTx
+        pendingTx.signers[msg.sender] = true;
+        pendingTx.nSigned += 1;
```

Comment:
nothing to worry about for now, just wanted to share. i've heard using a counter can potentially save gas, haven't played around with it myself so not sure...  could be a fun comparison

Reply:
Huh interesting - I looked at the open zeppelin implementation for counter and it used just a uint in itself so I wonder how it's different with gas - definitely will come back to it, as well as how to write code to save gas in general.

Reply:
I think it has to do with `unchecked` saving gas (which is used in the Counters implementation), can't find the exact reference, but `C44` here is a start https://github.com/Rari-Capital/solcurity